services/static.js

```diff
@@ -179,13 +202,15 @@ Static.getStaticImage = function(config) {
       v.oneOfType(
         v.oneOf('auto'),
         v.strictShape({
-          coordinates: v.required(v.coordinates),
```

Comment:
I think we could improve this, right now it will let you pass an empty object for position.

Instead could we do oneOfType:
- coordinates, zoom required, rest optional
- bbox require, rest optional

Reply:
thanks for your review, let me do some testing and get back to you!

Reply:
@andrewharvey, Can you explain what do you mean by "it will let you pass an empty object?" When I test this on my side with `position: {}`, I receive the following error: 
```
../mapbox-sdk-js/services/static.js:323
    .concat([
     ^
TypeError: Cannot read property 'concat' of undefined
```
Even though this is not a specific e error message but it does not pass beyond the `encodePosition()` function. Moving forward I can add an error to catch this scenario and return a more helpful message inside the encodePosition()` function. 
Another option here is to configure the position as you suggested:    
```
position: v.required(
v.oneOfType(
        v.oneOf('auto'),
        v.strictShape({
          coordinates: v.required(v.coordinates)
          zoom: v.required(v.range([0, 20])),
          bearing: v.range([0, 360]),
          pitch: v.range([0, 60]),
          bbox: v.arrayOf(v.number)
        }),
        v.strictShape({bbox: v.required(v.arrayOf(v.number))})
      )
```
With this method, we are relying on the[ fusspot library](https://github.com/mapbox/fusspot#vnumber) to validate the position object. In `position:{}` scenario , fusspot will return an error `Error: position.bbox is required` which is not necessarily true because only `positon` is required and it could be in a form of `bbox`,`auto` or `coordinates & zoom`. In case of an additional location parameter with the bbox position such as: `position: { bbox:[-77.04,38.89,-77.02,38.91], pitch: 4}` , the fusspot will simply return a an error related to the first `oneOfType()` arguments. In this case: `Error: position must be "auto"` which is not necessarily helpful either. 

Let me know if this is helpful or if you have any other suggestions!

Reply:
Another thing to consider is that it is not possible to use the `bbox` with any other location parameters such as `zoom`, `pitch`, and `bearing`.